tests/Doctrine/Tests/ORM/Functional/Ticket/GH6531Test.php

```diff
@@ -0,0 +1,242 @@
+<?php
+
+use Doctrine\Common\Collections\ArrayCollection;
+
+/**
+ * @group GH-6531
+ */
+class GH6531Test extends \Doctrine\Tests\OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setup();
+
+        try {
+            $this->_schemaTool->createSchema(array(
+                $this->_em->getClassMetadata(__NAMESPACE__ . '\GH6531User'),
+                $this->_em->getClassMetadata(__NAMESPACE__ . '\GH6531Address'),
+                $this->_em->getClassMetadata(__NAMESPACE__ . '\GH6531Article'),
+                $this->_em->getClassMetadata(__NAMESPACE__ . '\GH6531ArticleAttribute'),
+                $this->_em->getClassMetadata(__NAMESPACE__ . '\GH6531Customer'),
+                $this->_em->getClassMetadata(__NAMESPACE__ . '\GH6531Order'),
+                $this->_em->getClassMetadata(__NAMESPACE__ . '\GH6531OrderItem'),
+                $this->_em->getClassMetadata(__NAMESPACE__ . '\GH6531Product'),
+            ));
+        } catch (\Exception $e) {
+            return;
+        }
+    }
+
+    /**
+     * @group GH-6531
+     */
+    public function testSimpleDerivedIdentity(): void
+    {
+        $user = new GH6531User();
+        $address = new GH6531Address();
+        $address->setUser($user);
+
+        $this->_em->persist($user);
+        $this->_em->persist($address);
+        $this->_em->flush();
+    }
+
+
+    /**
+     * @group GH-6531
+     */
+    public function testDynamicAttributes(): void
+    {
+        $article = new GH6531Article('An Article');
+        $article->addAttribute('name', 'value');
+
+        $this->_em->persist($article);
+        $this->_em->flush();
+    }
+
+    /**
+     * @group GH-6531
+     */
+    public function testJoinTableWithMetadata(): void
+    {
+        $customer = new GH6531Customer();
+        $product = new GH6531Product('A Product', 10.0);
+        $this->_em->persist($customer);
+        $this->_em->persist($product);
+        $this->_em->flush();
+
+        $order = new GH6531Order($customer);
+        $order->addItem($product, 2);
+
+        $this->_em->persist($order);
+        $this->_em->flush();
+    }
+
+}
+
+/**
+ * @Entity
+ */
+class GH6531User
+{
+    /** @Id @Column(type="integer") @GeneratedValue */
+    private $id;
+}
+
+/**
+ * @Entity
+ */
+class GH6531Address
+{
+    /** @Id @OneToOne(targetEntity="GH6531User") */
+    private $user;
+
+    public function setUser($user)
+    {
+        $this->user = $user;
+    }
+}
+
+/**
+ * @Entity
+ */
+class GH6531Article
+{
+    /** @Id @Column(type="integer") @GeneratedValue */
+    private $id;
+
+    /** @Column(type="string") */
+    private $title;
```

Comment:
Remove any fields and methods that aren't strictly required for the test scenario